woocommerce/woocommerce
# Update nighty build workflow

### Changes proposed in this Pull Request:

Split from https://github.com/woocommerce/woocommerce/issues/51027

Updates the `nightly-builds.yml` workflow.

- update the tag using github-script instead of a third party action
- updated the WebFreak001/deploy-nightly action to a recent version, it was very outdated
- extract some env variables

### How to test the changes in this Pull Request:

The workflow also has a manual trigger. Trigger [the workflow](https://github.com/woocommerce/woocommerce/actions/workflows/nightly-builds.yml) from this branch. Then check the [Nightly release](https://github.com/woocommerce/woocommerce/releases/tag/nightly). It should have been updated.
Tested with this run: https://github.com/woocommerce/woocommerce/actions/runs/10834693750
